### PR TITLE
Publish preview jobs with preview release settings

### PR DESCRIPTION
**Pull request by gracechensd**
**Context:** Preview release artifacts (in particular jars for 6.0.0-ccs-alpha1) were not getting published to Artifactory. As a result, 6.0.x-alpha1 branches in `common` and other PLATFORM repos were not able to build successfully.

**Changes:** This commit updates the Jenkinsfile to handle preview release jobs. It checks if the job is a preview job (denoted by -alpha in the branch name), and use specific preview release settings for publishing to `maven-releases-preview` in Artifactory if so.
This was tested with replay option on [this 6.0.x-alpha1 job](https://jenkins.confluent.io/job/confluentinc/job/kafka/job/6.0.x-alpha1/17/) and it ran successfully, publishing artifacts to Artifactory and allowing platform repos to build (e.g. [successful build in common](https://jenkins.confluent.io/job/confluentinc/job/common/job/6.0.x-alpha1/10/)).

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -53,6 +53,12 @@ def job = {
               sh "./gradlew --init-script ${GRADLE_NEXUS_SETTINGS} --no-daemon uploadArchivesAll"
           }
       }
+    } else if (config.publish && config.isPreviewJob) {
```
Comment (elismaga) [inline Jenkinsfile]: i think this can be done with out duplicating the gradle command and the stage definition.

**Comment by gracechensd** [comment]
2 changes:

- Artifact publishing: Did some cleanup to minimize code duplication. I tested this version by running locally with the 6.0.x-alpha1 job and verified that it publishes to Artifactory.
- Added skip signing for alpha branches. This is a change that was introduced for 6.0.x-alpha1 in [this commit](https://github.com/confluentinc/kafka/commit/1a8bea870bbf78b654182090255823687810edf5) which has worked successfully for the preview job, and needs to be backported to master.

File: Jenkinsfile [hunk 2]
```diff
@@ -47,11 +52,13 @@ def job = {
...
-      configFileProvider([configFile(fileId: 'Gradle-Artifactory-Settings', variable: 'GRADLE_NEXUS_SETTINGS')]) {
-          stage("Publish to artifactory") {
-              sh "./gradlew --init-script ${GRADLE_NEXUS_SETTINGS} --no-daemon uploadArchivesAll"
-          }
+    if (config.publish) {
+      stage("Publish to artifactory") {
+        if (config.isDevJob) {
+          publishStep('Gradle-Artifactory-Settings')
+        } else if (config.isPreviewJob) {
+          publishStep('Gradle-Artifactory-Preview-Release-Settings')
```
Comment (mohnishbasha) [inline Jenkinsfile]: this is neat +1

File: build.gradle [hunk 3]
```diff
@@ -123,7 +123,7 @@ ext {
   userMaxTestRetryFailures = project.hasProperty('maxTestRetryFailures') ? maxTestRetryFailures.toInteger() : 0
 
   skipSigning = project.hasProperty('skipSigning') && skipSigning.toBoolean()
-  shouldSign = !skipSigning && !version.endsWith("SNAPSHOT") && project.gradle.startParameter.taskNames.any { it.contains("upload") }
+  shouldSign = !skipSigning && !version.endsWith("SNAPSHOT") && !version.contains("-alpha") && project.gradle.startParameter.taskNames.any { it.contains("upload") }
```
Comment (mohnishbasha) [inline build.gradle]: can we also find out if we need to skip signing for -beta as well?
- do not need to do this as of now, can be done later.

Comment (gracechensd) [reply build.gradle]: thats a good point, will note in a ticket 👍

**Pull request merged**